src/rate.rs

```diff
@@ -726,7 +726,8 @@ impl RCState {
   }
 
   pub fn update_state(
-    &mut self, bits: i64, fti: usize, log_target_q: i64, droppable: bool
+    &mut self, bits: i64, fti: usize, log_target_q: i64, trial: bool,
+    droppable: bool
   ) -> bool {
     let mut dropped = false;
     // Update rate control only if rate control is active.
```

Comment:
Is it possible that self.target_bitrate <= 0 when update_state() is called with trial == true?  If so the else side of this block will not execute and your trial encode will not be recorded.  Do we want an assert!() for this to catch changed behavior?

Reply:
So, if `trial` implies `needs_trial_encode()` then this cannot be the case. Let's keep the asserts that were dropped.